server/src/main/java/com/paypal/selion/grid/servlets/GridAutoUpgradeDelegateServlet.java

```diff
@@ -160,24 +160,12 @@ protected void process(HttpServletRequest request, HttpServletResponse response)
             }
 
             if (pendingProxy.size() > 0) {
-                writer.write("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">");
-                writer.write("<html xmlns=\"http://www.w3.org/1999/xhtml\">");
-                writer.write("<body id='main_body'>");
-                writer.write("<form id='myForm' name='myForm' class='appnitro' method='post' onsubmit='validateForm()' action='"
-                        + GridAutoUpgradeDelegateServlet.class.getSimpleName() + "' >");
                 String ids = "";
                 for (String temp : pendingProxy) {
                     ids = ids + temp + ",";
                 }
                 ids = StringUtils.chop(ids);
-                writer.write("The following nodes were not auto upgraded: " + ids);
-                writer.write("<br>Click the Submit button to retry.");
-                writer.write("<input type='hidden' name='" + IDS + "' value='" + ids + "'>");
-                writer.write("<input type='hidden' name='" + PARAM_JSON + "' value='" + downloadJSON + "'>");
-                writer.write("<input id='saveForm' class='button_text' type='submit' name='submit' value='Submit' />");
-                writer.write("</form>");
-                writer.write("</body>");
-                writer.write("</html>");
+                ServletHelper.respondWithTemplate(writer, PENDING_NODE_FILE, ids, ids, downloadJSON);
```

Comment:
I'd still like this to be further refactored. Rather than passing in the `writer`, I'd like us to pass the `response` object and all the input for consideration when writing the response. I'll go ahead and do the changes to this branch.